cheatsheets/DotNet_Security_Cheat_Sheet.md

```diff
@@ -609,20 +613,55 @@ public void RemoveAntiForgeryCookie(Controller controller)
 }
 ```
 
-NB: You will need to attach the anti-forgery token to Ajax requests.
+#### Using .NET Core 2.0 or later:
+
+After .NET Core 2.0 it is possible to automatically generate and verify the antiforgery token.
 
-After .NET Core 2.0 it is possible to automatically generate and verify the antiforgery token. Forms must have the requisite helper as seen here:
+If you are using tag-helpers - what is the default for most web project templates - all forms will automatically send the anti-forgery token.
+You can check if tag-helpers are enabled by checking if your main `_ViewImports.cshtml`- file contains:
+
+```csharp
+@addTagHelper *, Microsoft.AspNetCore.Mvc.TagHelpers
+```
+
+`IHtmlHelper.BeginForm` also sends anti-forgery-tokens automatically.
+
+If you are not using tag-helpers or `IHtmlHelper.BeginForm`, forms must have the requisite helper as seen here:
 
 ```html
 <form action="RelevantAction" >
 @Html.AntiForgeryToken()
 </form>
 ```
 
-And then add the `[AutoValidateAntiforgeryToken]` attribute to the action result.
+And then add the `[AutoValidateAntiforgeryToken]` attribute to the action method. This will validate the token for all HTTP methods other than GET, HEAD, OPTIONS and TRACE.
+
+#### Using .Net Core 2.0 or full .NET-Framework with AJAX
+
+You will need to attach the anti-forgery token to Ajax requests.
+
+If you are using jQuery in an ASP .NET Core MVC view this can be achieved using this snippet:
+
+```javascript
+@inject  Microsoft.AspNetCore.Antiforgery.IAntiforgery antiforgeryProvider
+$.ajax(
+{
+    type: "POST",
+    url: '@Url.Action("Action", "Controller")',
+    contentType: "application/x-www-form-urlencoded; charset=utf-8",
+    data: {
+        id: id,
+        '__RequestVerificationToken': '@antiforgeryProvider.GetAndStoreTokens(this.Context).RequestToken'
+    }
+})
+```
+
+If you are using the full .NET-Framework you can find some snippets [here](https://docs.microsoft.com/de-de/aspnet/web-api/overview/security/preventing-cross-site-request-forgery-csrf-attacks#anti-csrf-and-ajax).
```

Comment:
Change this to the English version (`en-us` rather than `de-de` in the URL).

Reply:
Fixed.